src/reader.rs

```diff
@@ -637,44 +626,34 @@ impl<R: Read> ReadPgn for BufferedReader<R> {
     type Err = io::Error;
 
     fn fill_buffer_and_peek(&mut self) -> io::Result<Option<u8>> {
```

Comment:
Correctness requires that this method definitely fills the buffer with at least `MIN_BUFFER_SIZE` bytes, if available. Is this true with the new implementation?

Reply:
It loops until available_data in the buffer is MIN_BUFFER_SIZE and it only breaks out of the loop when it reads no more data, which is the same logic as was used with slice-deque, so yes that behaviour is unchanged.

To be honest when changing that function I was unsure why it needed to read to fill the buffer that far. It can't be for correctness in parsing as far as I understand it there is no reason to see that far ahead. And it's unnecessary for safety reasons (the buffer gets fully initialized at the very beginning and constantly reused, so there are never any poisonous undefined bytes that can get read).

Reply:
It needs this only in a one or two places, parsing headers and comments, so that the entire header/comment will be present in the slice at once. All other places could use a more relaxed "peek", if there's a performance difference.

Is it not possible for `space()` to be empty, and then `read()` being called on a 0-sized buffer and breaking out of the loop?

Reply:
Unless the consume_noshift variant of consume is used the buffer will be shifted when the position pointer of the buffer is over the halfway point in it's internal storage. 

That means that available_data() + available_space() should always be more than half of the buffer size which I kept at MIN_BUFFER_SIZE \* 2.

Reply:
Ah, cool.